core/src/test/java/info/rmapproject/core/rmapservice/impl/openrdf/ORMapDiSCOMgrTest.java

```diff
@@ -299,6 +305,63 @@ public void testCreateAndUpdateDiSCOWithBNodes() throws RMapException, RMapDefec
 		}	
 	}
 	
+	/* Creating a DiSCO should define a lineage with that disco as progenitor */
+    @Test
+    public void testCreateDiscoLineage() throws Exception {
+        ORMapDiSCO disco = new ORMapDiSCO(uri2OpenRdfIri(randomURI()), 
+                new RMapIri(randomURI()), 
+                Arrays.asList(randomURI()));
+
+        discomgr.createDiSCO(disco, reqEventDetails, triplestore);
+
+        assertEquals(disco.getId().getIri(), findLineageProgenitor(disco.getId().getIri(), triplestore));
+    }
+
+	/* Updates (by the same agent) should all have the same lineage progenitor */
+    @Test
+    public void testUpdateDiscoLineage() {
+
+        final ORMapDiSCO originalDisco = new ORMapDiSCO(uri2OpenRdfIri(randomURI()), 
+                new RMapIri(randomURI()), 
+                Arrays.asList(randomURI()));
+
+        final ORMapDiSCO updatedDisco = new ORMapDiSCO(uri2OpenRdfIri(randomURI()), 
+                new RMapIri(randomURI()), 
+                Arrays.asList(randomURI()));
+
+        discomgr.createDiSCO(originalDisco, reqEventDetails, triplestore);
+
+        discomgr.updateDiSCO(
+                rMapIri2OpenRdfIri(originalDisco.getId()), updatedDisco, reqEventDetails, false, triplestore);
+
+        assertEquals(findLineageProgenitor(originalDisco.getId().getIri(), triplestore),
```

Comment:
Might be worthwhile to add one more discomgr.updateDiSCO so that we now have 3 DiSCO IDs to ensure the first discoId is still taken.